spanner/api/Spanner.Samples/DeleteInstanceConfigAsync.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright 2022 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_delete_instance_config]
+
+using System;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.Admin.Instance.V1;
+using Grpc.Core;
+
+public class DeleteInstanceConfigAsyncSample
+{
+
+    public async Task DeleteInstanceConfigAsync(string projectId, string instanceConfigId)
+    {
+        InstanceAdminClient instanceAdminClient = await InstanceAdminClient.CreateAsync();
+        var instanceConfigName = new InstanceConfigName(projectId, instanceConfigId);
+
+        //Check if there is a custom config with this name.
+        await instanceAdminClient.GetInstanceConfigAsync(instanceConfigName);
```

Comment:
Instead of trying to obtain the config here (this will fail if the config does not exist), I think it's better to wrap the delete operation in a try catch and look for an RpcException with status "NOT_FOUND", which means the config didn't exist.